File: client-clojure/src/sirqul_io_t_platform/specs/itinerary_response.clj
(ns sirqul-io-t-platform.specs.itinerary-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.vehicle-response :refer :all]
            [sirqul-io-t-platform.specs.stop-response :refer :all]
            )
  (:import (java.io File)))


(def itinerary-response-data
  {
   (ds/opt :itineraryId) string?
   (ds/opt :waypointCount) int?
   (ds/opt :score) int?
   (ds/opt :errorCode) string?
   (ds/opt :error) string?
   (ds/opt :itemCount) int?
   (ds/opt :vehicle) vehicle-response-spec
   (ds/opt :stops) (s/coll-of stop-response-spec)
   (ds/opt :estimatedStartTime) int?
   (ds/opt :estimatedEndTime) int?
   (ds/opt :estimatedDuration) int?
   (ds/opt :estimatedDistance) float?
   (ds/opt :waitTime) int?
   (ds/opt :generated) boolean?
   (ds/opt :valid) boolean?
   })

(def itinerary-response-spec
  (ds/spec
    {:name ::itinerary-response
     :spec itinerary-response-data}))

File: client-scalaz/src/main/scala/org/openapitools/client/api/ItineraryResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import ItineraryResponse._

case class ItineraryResponse (
  itineraryId: Option[String],
waypointCount: Option[Long],
score: Option[Long],
errorCode: Option[String],
error: Option[String],
itemCount: Option[Integer],
vehicle: Option[VehicleResponse],
stops: Option[List[StopResponse]],
estimatedStartTime: Option[Long],
estimatedEndTime: Option[Long],
estimatedDuration: Option[Long],
estimatedDistance: Option[Double],
waitTime: Option[Long],
generated: Option[Boolean],
valid: Option[Boolean])

object ItineraryResponse {
  import DateTimeCodecs._

  implicit val ItineraryResponseCodecJson: CodecJson[ItineraryResponse] = CodecJson.derive[ItineraryResponse]
  implicit val ItineraryResponseDecoder: EntityDecoder[ItineraryResponse] = jsonOf[ItineraryResponse]
  implicit val ItineraryResponseEncoder: EntityEncoder[ItineraryResponse] = jsonEncoderOf[ItineraryResponse]
}
